implementations/rust/ockam/ockam_api/src/cli_state/storage/projects_repository_sql.rs

```diff
@@ -44,22 +44,61 @@ impl ProjectsSqlxDatabase {
     pub async fn create() -> Result<Self> {
         Ok(Self::new(SqlxDatabase::in_memory("projects").await?))
     }
+
+    async fn is_shared_project(
+        &self,
+        project: &ProjectModel,
+        transaction: &mut AnyConnection,
+    ) -> Result<bool> {
+        // There are three possible scenarios:
+        // 1. The user enrolls and gets a project as an admin. There is an email from the "user"
+        //      table that matches an email in the project's "user_role" field that has role "admin".
+        // 2. Someone shares a project to the user via an invitation. There is an email from the "user"
+        //      table that matches an email in the project's "user_role" field that has role "service_user".
+        // 3. The user uses a ticket to enroll into a project. There is no email from the "user" table
+        //      that matches an email in the project's "user_role" field.
+        // This function returns true if we are in the second scenario; false otherwise.
+
+        // Get emails from user_roles that are not admins
+        let mut non_admin_emails = vec![];
+        for user_role in &project.user_roles {
+            if user_role.role != RoleInShare::Admin {
+                non_admin_emails.push(user_role.email.to_string());
+            }
+        }
+
+        // Check if any of the emails are in the user table
+        let q = query_scalar(r#"SELECT EXISTS(SELECT 1 FROM "user" WHERE email IN ($1))"#)
+            .bind(non_admin_emails.join(","));
+        let shared: Boolean = q.fetch_one(transaction).await.into_core()?;
+        Ok(shared.to_bool())
+    }
 }
 
 #[async_trait]
 impl ProjectsRepository for ProjectsSqlxDatabase {
     async fn store_project(&self, project: &ProjectModel) -> Result<()> {
         let mut transaction = self.database.begin().await.into_core()?;
 
-        // Is there a default project already?
-        let query1 = query("SELECT project_id FROM project WHERE is_default = $1").bind(true);
-        let project_id: Option<String> = query1
-            .fetch_optional(&mut *transaction)
-            .await
-            .into_core()?
-            .map(|row| row.get(0));
-        // The project is set as the default one if no other default project exists already
-        let is_default = project_id.is_none() || project_id == Some(project.id.clone());
+        let is_default: bool;
+        let mut project_name = &project.name;
+        // If it's a shared project, it can't be set as default.
+        if self.is_shared_project(project, &mut transaction).await? {
+            is_default = false;
+            // Also, the name is set to the project id  to avoid collisions with other
+            // projects with the same name that belong to other spaces.
+            project_name = &project.id;
```

Comment:
This is out of this PR scope but we definitely need to come back to this because we arbitrarily change a project name for the user. This might be surprising for any UI that put on top:

 - You've been invited to the "erics-services" project
 - "show services" --> the services shared by project 8195bd43-eb4f-4e4a-9650-8fed1cc028a5 are ...

One way to solve this would be to accept the ambiguity and return an error whenever the project name is ambiguous. Then ask the user to specify the space name or space id.

Reply:
Agree. We should also stop using space/project names outside the command layer. The rest of the codebase should just work with their IDs. That should also help to remove this issue.